data-pipeline-ffi/src/trace_exporter.rs

```diff
@@ -1,88 +1,230 @@
 // Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
 // SPDX-License-Identifier: Apache-2.0
 
+use crate::error::{ExporterError as Error, ExporterErrorCode as ErrorCode};
+use data_pipeline::trace_exporter::agent_response::AgentResponse;
 use data_pipeline::trace_exporter::{
-    ResponseCallback, TraceExporter, TraceExporterInputFormat, TraceExporterOutputFormat,
+    TraceExporter, TraceExporterInputFormat, TraceExporterOutputFormat,
 };
 use ddcommon_ffi::{
-    slice::{AsBytes, ByteSlice},
-    CharSlice, MaybeError,
+    CharSlice,
+    {slice::AsBytes, slice::ByteSlice},
 };
-use std::{ffi::c_char, ptr::NonNull, time::Duration};
+use std::{ptr::NonNull, time::Duration};
+
+/// The TraceExporterConfig object will hold the configuration properties for the TraceExporter.
+/// Once the configuration is passed to the TraceExporter constructor the config is no longer
+/// needed by the handle and it can be freed.
+#[derive(Default, PartialEq)]
+pub struct TraceExporterConfig {
+    url: Option<String>,
+    tracer_version: Option<String>,
+    language: Option<String>,
+    language_version: Option<String>,
+    language_interpreter: Option<String>,
+    hostname: Option<String>,
+    env: Option<String>,
+    version: Option<String>,
+    service: Option<String>,
+    input_format: TraceExporterInputFormat,
+    output_format: TraceExporterOutputFormat,
+    compute_stats: bool,
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn ddog_trace_exporter_config_new(
+    out_handle: NonNull<Box<TraceExporterConfig>>,
+) {
+    out_handle
+        .as_ptr()
+        .write(Box::<TraceExporterConfig>::default());
+}
+
+/// Frees TraceExporterConfig handle internal resources.
+#[no_mangle]
+pub unsafe extern "C" fn ddog_trace_exporter_config_free(handle: Box<TraceExporterConfig>) {
+    drop(handle);
+}
+
+/// Sets traces destination.
+#[no_mangle]
+pub unsafe extern "C" fn ddog_trace_exporter_config_set_url(
+    config: ddcommon_ffi::Option<&mut TraceExporterConfig>,
+    url: CharSlice,
+) -> ddcommon_ffi::Option<Error> {
+    if let ddcommon_ffi::Option::Some(handle) = config {
+        handle.url = Some(url.to_utf8_lossy().to_string());
+        None.into()
+    } else {
+        let code = ErrorCode::InvalidArgument;
+        ddcommon_ffi::Option::Some(Error::new(code, &code.to_string()))
+    }
+}
+
+/// Sets tracer's version to be included in the headers request.
+#[no_mangle]
+pub unsafe extern "C" fn ddog_trace_exporter_config_set_tracer_version(
+    config: ddcommon_ffi::Option<&mut TraceExporterConfig>,
+    version: CharSlice,
+) -> ddcommon_ffi::Option<Error> {
+    if let ddcommon_ffi::Option::Some(handle) = config {
+        handle.tracer_version = Some(version.to_utf8_lossy().to_string());
```

Comment:
Should we be using `to_utf8_lossy()` for stuff like this? `to_utf8_lossy` replaces invalid utf8 characters with a � so the string is technically valid utf8, but the value is basically corrupted for its intended purposes. Would it be better to use `try_to_utf8()` and return an error here? 

This feels like one of those edgecases that if it does become a bug, will be hard to track down later. Or maybe I'm just being paranoid. (If you agree, we can address in a follow-up PR).